sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/SqlAnalyzer.java

```diff
@@ -84,15 +84,19 @@ static Builder withQueryParams(Map<String, Value> params) {
    * resolution strategy set in the context.
    */
   ResolvedStatement analyze(String sql) {
-    AnalyzerOptions options = initAnalyzerOptions(builder.queryParams);
+    AnalyzerOptions options = initAnalyzerOptions();
+    for (Map.Entry<String, Value> entry : builder.queryParams.entrySet()) {
```

Comment:
I am slightly against this move because the original intention is to initialize `AnalyzerOptions` in the same place to avoid many pieces of initializations (which will be hard to maintain). I meant "slightly" because I can see why you are moving it: it will require static parameters as input.


What do you think about the idea of having a AnalyzerOptions builder with default settings on fields?

Reply:
I agree that we should have a single place that performs common init of AnalizerOptions and I think this change still achieves that goal. The query parameters are only set in the analyze path, so I think it actually creates more confusion to move it into a separate function. I have no strong opinions and can add a function if you'd like one.

Reply:
Ok maybe just a function and we can leave more work for future.

Reply:
Done. Please take another look.